wire-ios-sync-engine/Source/UserSession/ZMUserSession.swift

```diff
@@ -696,17 +696,19 @@ extension ZMUserSession: ZMSyncStateDelegate {
 
         let selfClient = ZMUser.selfUser(in: syncContext).selfClient()
 
-        if selfClient?.hasRegisteredMLSClient == true {
-            mlsService.performPendingJoins()
-            mlsService.uploadKeyPackagesIfNeeded()
-            mlsService.updateKeyMaterialForAllStaleGroupsIfNeeded()
-            commitPendingProposalsIfNeeded()
-        }
-        fetchFeatureConfigs()
-        recurringActionService.performActionsIfNeeded()
+        WaitingGroupTask(context: syncContext) { [self] in
```

Comment:
@typfel Do you think it's okay to execute this code async?

Reply:
Should I change it to have only four statements in the waiting group? @typfel 

<img width="1442" alt="Screenshot 2023-12-14 at 10 22 41" src="https://github.com/wireapp/wire-ios/assets/15628584/142387c6-a171-417d-a7ae-f1fe1e21d43c">

Reply:
> Should I change it to have only four statements in the waiting group? @typfel

Yes, I think that make sense and you could call `try await mlsService.commitPendingProposals()` directly. `mlsService.performPendingJoins` I think we want to execute on the syncMOC since it access and modifies local state.